cristin-project/src/main/java/no/unit/nva/cristin/projects/ProjectPatchValidator.java

```diff
@@ -0,0 +1,141 @@
+import static no.unit.nva.cristin.model.JsonPropertyNames.ALTERNATIVE_TITLES;
+import static no.unit.nva.cristin.model.JsonPropertyNames.CONTRIBUTORS;
+import static no.unit.nva.cristin.model.JsonPropertyNames.COORDINATING_INSTITUTION;
+import static no.unit.nva.cristin.model.JsonPropertyNames.END_DATE;
+import static no.unit.nva.cristin.model.JsonPropertyNames.LANGUAGE;
+import static no.unit.nva.cristin.model.JsonPropertyNames.POPULAR_SCIENTIFIC_SUMMARY;
+import static no.unit.nva.cristin.model.JsonPropertyNames.START_DATE;
+import static no.unit.nva.cristin.model.JsonPropertyNames.TITLE;
+import static no.unit.nva.language.LanguageConstants.UNDEFINED_LANGUAGE;
+import static no.unit.nva.language.LanguageMapper.getLanguageByIso6391Code;
+import static no.unit.nva.language.LanguageMapper.getLanguageByUri;
+import static nva.commons.core.attempt.Try.attempt;
+
+
+public class ProjectPatchValidator {
+
+    public static final String FIELD_CAN_NOT_BE_ERASED = "Field %s can not be erased";
+    public static final String ILLEGAL_VALUE_FOR_PROPERTY = "Illegal value for '%s'";
+
+    /**
+     * Validate changes to Project, both nullable fields and values.
+     * @param input ObjectNode containing fields with input data to be changed.
+     * @throws BadRequestException thrown when input has illegal or invalid values.
+     */
+    public static void validate(ObjectNode input) throws BadRequestException {
+
+        validateRequiredFieldsNotNull(input);
+
+        validateLanguageStringMap(input, ACADEMIC_SUMMARY);
+        validateAlternativeTitles(input);
+        validateLanguageStringMap(input, POPULAR_SCIENTIFIC_SUMMARY);
+        validateInstant(input, END_DATE);
+        validateInstant(input, START_DATE);
+        validateLanguage(input);
+        validateStatus(input);
+        validateContributors(input);
+    }
+
+    private static void validateRequiredFieldsNotNull(ObjectNode input) throws BadRequestException {
+        validateNotNullIfPresent(input, TITLE);
+        validateNotNullIfPresent(input, START_DATE);
+        validateNotNullIfPresent(input, CONTRIBUTORS);
+        validateNotNullIfPresent(input, COORDINATING_INSTITUTION);
+    }
+
+    private static void validateContributors(ObjectNode input) throws BadRequestException {
+        TypeReference<List<NvaContributor>> typeRef = new TypeReference<>() {
+        };
+        try {
+            final String content = input.get(CONTRIBUTORS).asText();
+            List<NvaContributor> contributors = OBJECT_MAPPER.readValue(content, typeRef);
+            if (contributors.isEmpty()) {
+                throw new BadRequestException(String.format(FIELD_CAN_NOT_BE_ERASED, CONTRIBUTORS));
+            }
+        } catch (JsonProcessingException e) {
+            throw new BadRequestException(String.format(ILLEGAL_VALUE_FOR_PROPERTY, CONTRIBUTORS));
+        }
+    }
+
+    private static void validateStatus(ObjectNode input) throws BadRequestException {
```

Comment:
https://api.cristin-test.uio.no/v2/doc/json-schemas/projects_PATCH_request.json

Status is not a valid patch field. It gets calculated from dates